tools/tplg_parser/pga.c

```diff
@@ -93,16 +95,28 @@ int tplg_create_pga(struct tplg_context *ctx, struct sof_ipc_comp_volume *volume
 			return -EINVAL;
 		}
 
+		/* parse uuid token */
+		ret = sof_parse_tokens(uuid, comp_ext_tokens,
+				       ARRAY_SIZE(comp_ext_tokens), array,
+				       array->size);
+		if (ret != 0) {
+			fprintf(stderr, "error: parse pga uuid token %d\n", size);
+			free(array);
+			return -EINVAL;
+		}
+
 		total_array_size += array->size;
 	}
 
 	/* configure volume */
-	volume->comp.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_COMP_NEW;
-	volume->comp.id = comp_id;
-	volume->comp.hdr.size = sizeof(struct sof_ipc_comp_volume);
-	volume->comp.type = SOF_COMP_VOLUME;
-	volume->comp.pipeline_id = ctx->pipeline_id;
+	comp->hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_COMP_NEW;
+	comp->id = comp_id;
+	comp->hdr.size = sizeof(struct sof_ipc_comp_volume) + UUID_SIZE;
+	comp->type = SOF_COMP_VOLUME;
+	comp->pipeline_id = ctx->pipeline_id;
+	comp->ext_data_length = UUID_SIZE;
 	volume->config.hdr.size = sizeof(struct sof_ipc_comp_config);
+	memcpy((uint8_t *)comp + sizeof(*volume), &uuid, UUID_SIZE);
```

Comment:
`memcpy(volume + 1,...)`

Reply:
or better still a convenience macro for UUID copying (since I assume the relative position is calculated the same and the size is const)